internal/compiler/fileloader.go

```diff
@@ -463,6 +480,52 @@ func (p *fileLoader) createSyntheticImport(text string, file *ast.SourceFile) *a
 	return externalHelpersModuleReference
 }
 
+func (p *fileLoader) pathForLibFile(name string) string {
+	if cached, ok := p.pathForLibFileCache.Load(name); ok {
+		return cached
+	}
+
+	path := tspath.CombinePaths(p.defaultLibraryPath, name)
+	if p.opts.Config.CompilerOptions().LibReplacement.IsTrue() {
+		libraryName := getLibraryNameFromLibFileName(name)
+		resolveFrom := getInferredLibraryNameResolveFrom(p.opts.Config.CompilerOptions(), p.opts.Host.GetCurrentDirectory(), name)
```

Comment:
i think you want to store these resolutions so we can watch these locations ?

Reply:
Ah, yes!

Reply:
I did something, though I'm not sure if it's too hacky. Probably not?